tests/test_pay.py

```diff
@@ -3460,9 +3460,22 @@ def test_mpp_interference_2(node_factory, bitcoind, executor):
         wait_for(lambda: len(l2.rpc.listchannels(channel)['channels']) == 2)
         wait_for(lambda: len(l3.rpc.listchannels(channel)['channels']) == 2)
 
-    # Buyers check out some purchaseable stuff from the merchant.
-    i2 = l1.rpc.invoice(unit * 6, 'i2', 'i2')['bolt11']
-    i3 = l1.rpc.invoice(unit * 6, 'i3', 'i3')['bolt11']
+    # At this point, we have the following incoming channel capacities:
+    # 74094000, 52314000, 30318000, 19318000
+
+    # We *always* rotate through, since we have no published channels,
+    # but we can select badly and get an overlap. e.g. first invoice
+    # takes 30318000, 19318000 and 74094000.  Second will then take
+    # 52314000, and have to reuse 30318000, which gets exhausted by the
+    # first payer, thus leaving them unable to pay 66000000.
+
+    # So we re-do this until we only have 4 or fewer routehints.
+    while True:
```

Comment:
In principle this should be fixed in our running code, not in our test.

Suggest tagging a FIXME in our actual code pointing to this problem and say we have a minor sub-goal to remove this code in the test --- in principle we *should* eventually switch to something as smart as an estimator that will make this test pass without this extra code.

Reply:
I wasn't sure quite how to fix this.  We *could* run a fuzzy estimator of how much is remaining, but in practice that seems overkill for something which is so marginal.

Reply:
Hmm.  If we agree that the issue is marginal (i.e. the issue only matters when the total incoming invoices is at or almost at the total incoming capacity) then maybe the proper test-fix is to increase the incoming channel capacities instead so that most of the invoices generated across all runs have 2 or fewer routehints.

Reply:
i.e. if we are going to say "if the total value of invoices you are going to issue is near to the total incoming capacity you have, try increasing your incoming capacity instead", then the proper thing to do in the test is to increase the incoming capacities of the channels going to the payee.